app/assets/javascripts/spritegroup.js

```diff
@@ -0,0 +1,31 @@
+function SpriteGroup (spriteName, spriteGroup, objectPool, numSprite, spriteOriginPosX, spriteOriginPosY) {
+  this.spriteName = spriteName;
+  this.spriteGroup = spriteGroup;
+  this.numSprite = numSprite;
+  this.spriteOriginPosX = spriteOriginPosX;
+  this.spriteOriginPosY = spriteOriginPosY;
+
+  this.renderSprites = function (shape) {
+    var spriteFileName = "";
+    var spritePosX = spriteOriginPosX;
+    var spritePosY = spriteOriginPosY;
+
+    // todo: renders in an octagonal fashion
+
+    // renders in a box like fashion
+    if (shape == "box") {
+      console.log(objectPool.allObjects.length);
+      for (var index in objectPool.allObjects) {
+        if (objectPool.allObjects[index].id > 0) {
+          spriteFileName = spriteName + objectPool.allObjects[index].id;
```

Comment:
Naming convention, ie: die1, die2, ..., die6. See the names of the assets under `preload()` for `game.js`. Another reason why I switched `dice.value` to `dice.id` under `dice.js`.